chatbot-core/backend/app/models/chat.py

```diff
@@ -0,0 +1,255 @@
+    ASSISTANT = "assistant"
+
+
+class ChatMessageRequestType(str, Enum):
+    """
+    Enumeration of message types in a chat session.
+    """
+
+    NEW = "new"
+    REGENERATE = "regenerate"
+    EDIT = "edit"
+
+
+class ChatMessageErrorType(str, Enum):
+    """
+    Enumeration of possible error types in a chat message.
+    """
+
+    SYSTEM_ERROR = "system_error"
+    VALIDATION_ERROR = "validation_error"
+    NETWORK_ERROR = "network_error"
+    GENERATION_ERROR = "generation_error"
+
+
+class ChatSessionSharedStatus(str, Enum):
+    """
+    Enumeration of chat session sharing statuses.
+    """
+
+    PUBLIC = "public"
+    PRIVATE = "private"
+
+
+CHAT_MESSAGES_ID = "chat_messages.id"
+
+
+class ChatSession(Base):
+    """
+    Represents a chat session between a user and an agent.
+    Tracks conversation details, sharing status, and associated messages.
+    """
+
+    __tablename__ = "chat_sessions"
+
+    id: Mapped[UNIQUEIDENTIFIER] = mapped_column(UNIQUEIDENTIFIER(as_uuid=True), primary_key=True, default=uuid4)
+    user_id: Mapped[UNIQUEIDENTIFIER] = mapped_column(ForeignKey("users.id", ondelete="CASCADE"), nullable=False)
+    agent_id: Mapped[Optional[UNIQUEIDENTIFIER]] = mapped_column(ForeignKey("agents.id"), nullable=True)
+    description: Mapped[Optional[str]] = mapped_column(String(255), nullable=True)
+    one_shot: Mapped[bool] = mapped_column(Boolean, default=False)
+    shared_status: Mapped[ChatSessionSharedStatus] = mapped_column(
+        SQLAlchemyEnum(ChatSessionSharedStatus, native_enum=False), default=ChatSessionSharedStatus.PRIVATE
+    )
+    current_alternate_model: Mapped[Optional[str]] = mapped_column(String, nullable=True)
+    created_at: Mapped[datetime] = mapped_column(
+        DateTime(timezone=True), server_default=func.now(), default=lambda: datetime.now(timezone.utc)
+    )
+    updated_at: Mapped[datetime] = mapped_column(
+        DateTime(timezone=True), server_default=func.now(), onupdate=lambda: datetime.now(timezone.utc)
+    )
+    deleted_at: Mapped[Optional[datetime]] = mapped_column(DateTime(timezone=True), nullable=True, default=None)
+
+    # Define relationships
+    user: Mapped["User"] = relationship("User", back_populates="chat_sessions", cascade="save-update, merge")
+    agent: Mapped["Agent"] = relationship("Agent", back_populates="chat_sessions", cascade="save-update, merge")
+    chat_messages: Mapped[List["ChatMessage"]] = relationship(
+        "ChatMessage", back_populates="chat_session", cascade="all, delete-orphan", lazy="dynamic"
+    )
+
+
+class ChatMessage(Base):
+    """
+    Represents an individual message within a chat session.
+    Tracks message content, type, and associated metadata.
+    """
+
+    __tablename__ = "chat_messages"
+
+    id: Mapped[UNIQUEIDENTIFIER] = mapped_column(UNIQUEIDENTIFIER(as_uuid=True), primary_key=True, default=uuid4)
+    user_id: Mapped[UNIQUEIDENTIFIER] = mapped_column(ForeignKey("users.id", ondelete="CASCADE"), nullable=False)
```

Comment:
Because I want every query that related to the **chat** need to query along with current **user**. For e.g:

```python
chat_message = (
      self._db_session.query(ChatMessage)
      .filter(
           and_(
                ChatMessage.id == chat_message_id,
                ChatMessage.chat_session_id == chat_session_id,
                ChatMessage.user_id == user_id,
           )
       )
       .first()
)
return chat_message, None
```

We can replace by the join operation with the table `chat_session`. But I think that embed a field `user_id` into table `chat_message` is still good as we don't need to perform the join operation.

Reply:
> Because I want every query that related to the **chat** need to query along with current **user**. For e.g:
> 
> ```python
> chat_message = (
>       self._db_session.query(ChatMessage)
>       .filter(
>            and_(
>                 ChatMessage.id == chat_message_id,
>                 ChatMessage.chat_session_id == chat_session_id,
>                 ChatMessage.user_id == user_id,
>            )
>        )
>        .first()
> )
> return chat_message, None
> ```
> 
> We can replace by the join operation with the table `chat_session`. But I think that embed a field `user_id` into table `chat_message` is still good as we don't need to perform the join operation.

Ok, double check to make sure the data is consistent from chat sessions to chat messages

Reply:
Yes, already tested.